### ISSUE TITLE
Where Can I Report Security Vulnerability?

### ISSUE THREAD

**Issue by orangetw** [javaserverfaces/mojarra#4364]
I found a vulnerability in the latest version of Mojarra (vulnerable by default). But I can't find any security related mail from websites. 

Where can I report my finding?

**Comment by orangetw** [comment]
I tested the latest stable version 2.3.4. I also checked the master branch and seems the code not change too much. I will send my report in a few days!

Thanks!

**Comment by orangetw** [comment]
Seems this is a bot's action...

**Comment by orangetw** [comment]
@arjantijms OK, but I still don't get any confirm from him...

**Comment by javaserverfaces** [comment]
@orangetw sorry for late. yes, you can send the report to me and I also need to clarify how to handle with the vulnerability in the current context.

**Comment by orangetw** [comment]
@javaserverfaces @ren-zhijun-oracle 
Hi, I have send the report to above email! Thanks :)

**Comment by orangetw** [comment]
This is fixed in Mojarra 2.3.5, with patch https://github.com/eclipse-ee4j/mojarra/commit/1b434748d9239f42eae8aa7d37d7a0930c061e24

```
The getLocalePrefix function in ResourceManager.java in Eclipse Mojarra 2.3.4 is affected by Directory Traversal via the loc parameter. A remote attacker can download configuration files or Java bytecodes from applications.
```

**Comment by henrikplate** [comment]
I downloaded org.glassfish:javax.faces:2.3.5 and 2.3.6 from Maven Central, and they both contain a version of com.sun.faces.application.resource.ResourceManager.getLocalePrefix(FacesContext) that does not check for forbidden sequences in localePrefix. Can you confirm that those version are indeed fixed?

**Comment by henrikplate** [comment]
The fix implemented with eclipse-ee4j/mojarra@1b43474 modifies a branch condition in method com.sun.faces.application.resource.ResourceManager.getLocalePrefix(FacesContext). Data that seemingly comes from an Http request is checked for forbidden character sequences.

I had a look at the byte code of both versions 2.3.5 and 2.3.6, as available in Maven Central, and they do not include the above-mentioned check. Only version 2.3.7, released on August 22nd 2018, does include the modification. As such, it looks to me as if 2.3.5 and 2.3.6 are still vulnerable.

PS: Note that I just compared the code changes of the fix commit with the byte code of the different released versions. I did not check whether the vulnerability is actually exploitable in any of those versions, nor did I check whether the fix is complete.

**Comment by kguelzau** [comment]
@orangetw / @javaserverfaces is it true that also the 2.2.x branch is affected by this vulnerability?
This seems to match _all_ versions below 2.3.7:
https://nvd.nist.gov/vuln/detail/CVE-2018-14371/cpes?expandCpeRanges=true

And I can't see that the 2.2.x is affected:
https://github.com/javaserverfaces/mojarra/blob/166d2173544b40dcfe5359d8ecbb5dda9cb53385/jsf-ri/src/main/java/com/sun/faces/application/resource/ResourceManager.java#L590
